### ISSUE TITLE
NullReferenceException when configuring mapping `ToDictionaryWithValueType`

### ISSUE THREAD

**Issue by bslbckr** [agileobjects/AgileMapper#152]
This might be a follow up issue of #133 where some [advice](https://github.com/agileobjects/AgileMapper/issues/133#issuecomment-487026307) was given how to map to dictionaries. When we followed the advice, we received a NullReferenceException.

We have got the following data structures:
```csharp
namespace Source {
	public class Wrapper {
		public IDictionary<string, Data> Dict {get; private set;}
		
		public Wrapper (IList<string> inputs) {
			Dict = new Dictionary<string, Data>();
			foreach (string s in inputs) {
				Dict.Add(s, new Data{Value = s});
			}
		}
	}
	public class Data {
		public String Value {get; set;}
	}
}

namespace Target {
	public class Data {
		public String Value {get; set;}
	}
}
```

If we try to map `Source.Wrapper` to `IDictionary<string, Target.Data>` Mapper does not accept the configuration and instead throws a NullReferenceException:

```
[System.NullReferenceException: Object reference not set to an instance of an object.]
   at AgileObjects.AgileMapper.DataSources.ConfiguredDataSourceFactory.MembersConflict(UserConfiguredItemBase otherConfiguredItem)
   at AgileObjects.AgileMapper.Configuration.UserConfiguredItemBase.ConflictsWith(UserConfiguredItemBase otherConfiguredItem)
   at AgileObjects.AgileMapper.DataSources.ConfiguredDataSourceFactory.ConflictsWith(UserConfiguredItemBase otherConfiguredItem)
   at AgileObjects.AgileMapper.Configuration.UserConfigurationSet.<>c__DisplayClass108_0`2.<ThrowIfConflictingItemExists>b__0(TExistingItem ci)
   at AgileObjects.AgileMapper.Extensions.Internal.EnumerableExtensions.TryFindMatch[T](IList`1 items, Func`2 predicate, T& match)
   at AgileObjects.AgileMapper.Extensions.Internal.EnumerableExtensions.FirstOrDefault[T](IList`1 items, Func`2 predicate)
   at AgileObjects.AgileMapper.Configuration.UserConfigurationSet.ThrowIfConflictingItemExists[TConfiguredItem,TExistingItem](TConfiguredItem configuredItem, IList`1 existingItems, Func`3 messageFactory)
   at AgileObjects.AgileMapper.Configuration.UserConfigurationSet.Add(ConfiguredIgnoredMember ignoredMember)
   at AgileObjects.AgileMapper.Api.Configuration.MappingConfigurator`2.IgnoreMembersByFilter(Expression`1 memberFilter)
   at AgileObjects.AgileMapper.Api.Configuration.MappingConfigurator`2.IgnoreTargetMembersWhere(Expression`1 memberFilter)
   at Program.Main() :line 33
```

The Mapper-configuration looks as follows:
```csharp
Mapper.WhenMapping.From<Source.Wrapper>().ToDictionariesWithValueType<Target.Data>()
    .Map(ctxt => ctxt.Source.Dict).ToTarget()
    .And.IgnoreTargetMembersWhere(m => m.Name == "Dict");
```

The fiddle can be found here: https://dotnetfiddle.net/HNXFdh

**Comment by bslbckr** [comment]
Out of curiosity I just changed the mapping configuration from `.ToDictionariesWithValueType<Target.Data>()` to `.ToDictionaries.`. Unfortunately, this change didn't change the exception.

**Comment by SteveWilkes** [comment]
Thanks for reporting the issue - it was a bug in one of the configuration checks - it's fixed in the latest 1.5 release code. Release to foloow!

It actually works if you configure the `ToTarget` and member ignore [in the other order](https://dotnetfiddle.net/e6HI3J)!

**Comment by SteveWilkes** [comment]
This is fixed as of v1.5-preview3, which is now [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks again! :)
